src/Domain/CoordinateMaps/BulgedCube.hpp

```diff
@@ -0,0 +1,469 @@
+ *  0 & 0 & \eta\\
+ *  0 & 1 & 0\\
+ *  \end{bmatrix},\
+ *
+ *  B_{\zeta} =
+ *  \begin{bmatrix}
+ *  1 & 0 & 0\\
+ *  0 & 1 & 0\\
+ *  0 & 0 & \zeta\\
+ *  \end{bmatrix}\\
+ *
+ *  B_{\xi\eta} =
+ *  \begin{bmatrix}
+ *  0 & 0 & \xi\\
+ *  \eta & 0 & 0\\
+ *  0 & 1 & 0\\
+ *  \end{bmatrix},\
+ *
+ *  B_{\xi\zeta} =
+ *  \begin{bmatrix}
+ *  \xi & 0 & 0\\
+ *  0 & 1 & 0\\
+ *  0 & 0 & \zeta\\
+ *  \end{bmatrix},\
+ *
+ *  B_{\eta\zeta} =
+ *  \begin{bmatrix}
+ *  0 & 1 & 0\\
+ *  \eta & 0 & 0\\
+ *  0 & 0 & \zeta\\
+ *  \end{bmatrix}\\
+ *
+ *  B_{\xi\eta\zeta} =
+ *  \begin{bmatrix}
+ *  \xi & 0 & 0\\
+ *  0 & \eta & 0\\
+ *  0 & 0 & \zeta\\
+ *  \end{bmatrix}\f]
+ *
+ *  Now we can write the volume map \f$\phi\f$ in these terms:
+ *
+ * \f[\phi(\xi,\eta,\zeta) =\\
+ * B_{\zeta}
+ * \vec{\sigma}_{+\zeta}(\xi, \eta)+
+ * B_{\eta}
+ * \vec{\sigma}_{+\zeta}(\xi, \zeta)+
+ * B_{\xi}
+ * \vec{\sigma}_{+\zeta}(\eta, \zeta)-\\
+ * B_{\xi \eta}
+ * \vec{\sigma}_{+\zeta}(+1, \zeta)-
+ * B_{\xi \zeta}
+ * \vec{\sigma}_{+\zeta}(+1, \eta)+
+ * B_{\eta \zeta}
+ * \vec{\sigma}_{+\zeta}(+1, \xi))+\\
+ * B_{\xi \eta \zeta}
+ * \vec{\sigma}_{+\zeta}(+1, +1)
+ *  \f]
+ *
+ * ### The Bulged Sphere Map
+ * We now use the result above to provide the mapping for the bulged sphere.
+ * First we will define the variables \f$\rho_A\f$ and \f$\rho_{AB}\f$, for
+ * \f$A, B \in {\Xi,\mathrm{H}, \mathrm{Z}} \f$:
+ * \f[\rho_A = \sqrt{2 + A^2}\\
+ *    \rho_{AB} = \sqrt{1 + A^2 + B^2}\f]
+ * The final mapping is then:
+ * \f[\vec{x}(\xi,\eta,\zeta) = \frac{(1-s)r}{\sqrt{3}}
+ *  \begin{bmatrix}
+ *  \Xi\\
+ *  \mathrm{H}\\
+ *  \mathrm{Z}\\
+ *  \end{bmatrix} +
+ *  \frac{sr}{\sqrt{3}}
+ *  \begin{bmatrix}
+ *  \xi\\
+ *  \eta\\
+ *  \zeta\\
+ *  \end{bmatrix} + sr
+ *  \begin{bmatrix}
+ *  \xi & \Xi & \Xi\\
+ *  \mathrm{H} & \eta &\mathrm{H}\\
+ *  \mathrm{Z} & \mathrm{Z} & \zeta\\
+ *  \end{bmatrix}
+ *  \begin{bmatrix}
+ *  1/\rho_{\mathrm{H}\mathrm{Z}}\\
+ *  1/\rho_{\Xi\mathrm{Z}}\\
+ *  1/\rho_{\Xi\mathrm{H}}\\
+ *  \end{bmatrix} - sr
+ *  \begin{bmatrix}
+ *  \Xi & \xi & \xi\\
+ *  \eta & \mathrm{H} &\eta\\
+ *  \zeta & \zeta & \mathrm{Z}\\
+ *  \end{bmatrix}
+ *  \begin{bmatrix}
+ *  1/\rho_{\Xi}\\
+ *  1/\rho_{\mathrm{H}}\\
+ *  1/\rho_{\mathrm{Z}}\\
+ *  \end{bmatrix}\f]
+ *
+ */
+class BulgedCube {
+ public:
+  static constexpr size_t dim = 3;
+  explicit BulgedCube(double radius, double sphericity,
+                      bool with_equiangular_map);
+  BulgedCube() = default;
+  ~BulgedCube() = default;
+  BulgedCube(BulgedCube&&) = default;
+  BulgedCube(const BulgedCube&) = default;
+  BulgedCube& operator=(const BulgedCube&) = default;
+  BulgedCube& operator=(BulgedCube&&) = default;
+
+  template <typename T>
+  std::array<std::decay_t<tt::remove_reference_wrapper_t<T>>, 3> operator()(
+      const std::array<T, 3>& x) const noexcept;
+
+  template <typename T>
+  std::array<std::decay_t<tt::remove_reference_wrapper_t<T>>, 3> xi_derivative(
```

Comment:
what is this function and why is it public?

Reply:
helper function for computing the jacobian (will be made private)